python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -57,41 +67,98 @@ def __init__(self):
-        self.addParameter(ParameterSelection(self.RTYPE,
-                                             self.tr('Output raster type'), self.TYPE, 5))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_A,
+                self.tr('Input layer A'),
+                optional=False))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_A,
+                self.tr('Number of raster band for A'),
+                parentLayerParameterName=self.INPUT_A))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_B,
+                self.tr('Input layer B'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_B,
+                self.tr('Number of raster band for B'),
+                parentLayerParameterName=self.INPUT_B, 
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_C,
+                self.tr('Input layer C'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(self.BAND_C,
+                self.tr('Number of raster band for C'),
+                parentLayerParameterName=self.INPUT_C,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_D,
+                self.tr('Input layer D'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_D,
+                self.tr('Number of raster band for D'),
+                parentLayerParameterName=self.INPUT_D,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_E,
+                self.tr('Input layer E'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_E,
+                self.tr('Number of raster band for E'),
+                parentLayerParameterName=self.INPUT_E,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.INPUT_F, 
+                self.tr('Input layer F'),
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterBand(
+                self.BAND_F,
+                self.tr('Number of raster band for F'),
+                parentLayerParameterName=self.INPUT_F,
+                optional=True))
+        self.addParameter(
+            QgsProcessingParameterString(self.FORMULA,
+                self.tr('Calculation in gdalnumeric syntax using +-/* or any numpy array functions (i.e. logical_and())'), 
+                'A*2', 
+                optional=False))
+        self.addParameter(
```

Comment:
Should be a number input - copy https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/merge.py#L85